vsintegration/src/FSharp.Editor/FSharpColorizationService.fs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Collections.Concurrent
+open System.Collections.Generic
+open System.Linq
+open System.Threading
+open System.Threading.Tasks
+open System.Runtime.CompilerServices
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Classification
+open Microsoft.CodeAnalysis.Editor
+open Microsoft.CodeAnalysis.Editor.Implementation.Classification
+open Microsoft.CodeAnalysis.Editor.Shared.Utilities
+open Microsoft.CodeAnalysis.Host.Mef
+open Microsoft.CodeAnalysis.Text
+
+open Microsoft.VisualStudio.FSharp.LanguageService
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Tagging
+
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+// TODO: add types colorization if available from intellisense
+// TODO: add defines flags if available from project sites and files
+
+type internal SourceTextColorizationData(classificationData: seq<ClassifiedSpan>) =
+    member this.Tokens = classificationData |> Seq.toArray
```

Comment:
this is property -> you'll get fresh array allocation and walk over `classificationData` on every access. It seems that the only place where this type is used it passes `List<ClassifiedSpan>` as a constructor parameter so you can switch parameter to be `IReadOnlyList<ClassifiedSpan>` and then define `Tokens` as 
```fsharp
member val Tokens = classificationData
```